src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedDefaultTests.cs

```diff
@@ -54,6 +66,55 @@ class C
         }
 
         [Fact]
+        public void LambdaWithInference()
+        {
+            string source = @"
+public class D
+{
+    public static void Main()
+    {
+        new C<string, string>().M();
+    }
+}
+class C<T1, T2>
+{
+    public void M(bool b = true)
+    {
+        var map = new C<string, string>();
+        map.GetOrAdd("""", _ => default);
+        map.GetOrAdd("""", _ => null);
+
+        var map2 = new C<string, int>();
+        map2.GetOrAdd("""", _ => default);
+
+        var map3 = new C<string, (string, string)>();
+        map3.GetOrAdd("""", _ => default);
+        map3.GetOrAdd("""", _ => (null, null));
+
+        var map4 = new C<string, string>();
+        map4.GetOrAdd("""", _ => b switch { _ => null });
+        map4.GetOrAdd("""", _ => b switch { _ => default });
```

Comment:
should we have a test where the type arguments to GetOrAdd are unconstrained generics? (or maybe one already exists?)